app/assets/javascripts/admin/controllers/modals/admin-edit-badge-groupings.js.es6

```diff
@@ -1,16 +1,15 @@
+import BadgeGrouping from 'discourse/models/badge-grouping';
+
 export default Ember.Controller.extend({
   needs: ['modal'],
 
   modelChanged: function(){
-
-    var grouping = Em.Object.extend({});
-
     var model = this.get('model');
     var copy = Em.A();
 
     if(model){
       model.forEach(function(o){
-        copy.pushObject(grouping.create(o));
+        copy.pushObject(BadgeGrouping.create(o));
```

Comment:
It would be better if you used `this.store.createRecord('badge-grouping', o)` - then you don't have to import the model.